src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -86,9 +86,17 @@ export const mapColumn: ExpressionFunctionDefinition<
         .expression?.(...params)
         .pipe(take(1))
         .toPromise() ?? Promise.resolve(null);
-    const columnId = args.id != null ? args.id : args.name;
 
     const columns = [...input.columns];
+    const existingColumnIndex = columns.findIndex(({ id, name }) => {
+      if (args.id) {
+        return id === args.id;
+      }
+      return name === args.name;
+    });
+    const columnId =
+      existingColumnIndex === -1 ? args.id ?? args.name : columns[existingColumnIndex].id;
```

Comment:
ids need to be unique, if we find existing column with requested new column id we should throw

Reply:
imo the logic should be:

```
const columnId = args.id ?? args.name;
if(columns.find({id} => id === columnId)) throw('id already exists');
```

Reply:
@ppisljar Column IDs are guaranteed to be unique with this logic, I am not seeing any case where you could get a duplicate ID. Also, this logic lets you rewrite a column by ID, this is shown in some of the test cases. For example `mapColumn id="existing" name="New name" expression={math "existing + 2"}` is a valid expression that updates the data in place.